pkg/operations/kubernetesupgrade/upgradeagentnode.go

```diff
@@ -52,7 +52,8 @@ func (kan *UpgradeAgentNode) DeleteNode(vmName *string, drain bool) error {
 	var kubeAPIServerURL string
 
 	if kan.UpgradeContainerService.Properties.HostedMasterProfile != nil {
```

Comment:
Can you add unit test coverage for this and the below if/else as well?

Reply:
Not easily, no